### PR TITLE
fix(appConfig): restore check to app config

### PR DESCRIPTION
**Pull request by code-forger**
This validation was lost in the upgrade to webpack 5, restore it

### CODE REVIEW

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class RestrictRuntimeSymbols {
+  // eslint-disable-next-line class-methods-use-this -- no need for this
+  apply(compiler) {
+    compiler.hooks.assetEmitted.tap('RestrictRuntimeSymbols', (file, { targetPath }) => {
+      if (targetPath.endsWith('.node.js')) {
+        const initialContent = fs.readFileSync(targetPath, 'utf8');
+        if (initialContent.match(/create-react-class/)) {
+          throw new Error('`create-react-class` is restricted from being used');
+        }
+      } else if (targetPath.endsWith('rowser.js')) { // catch browser and legacyBrowser
```
Comment (dogpatch626) [inline packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:15]:
```js
 else if (targetPath.endsWith('rowser.js'))
```
did you mean 
```js
 else if (targetPath.endsWith('browser.js'))
```
?

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import fs from 'node:fs';
+import {
+  validateAppConfig,
+} from '@americanexpress/one-app-dev-bundler';
+
+class RestrictRuntimeSymbols {
+  // eslint-disable-next-line class-methods-use-this -- no need for this
+  apply(compiler) {
```
Comment (dogpatch626) [inline packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:8]: Where is compiler coming from and is this just passing the ```this``` value of compiler to apply?

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js (hunk 1 above)
Comment (code-forger) [reply packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:15]: No, as the comment mentions, `browser.js` does not match `legacyBrowser.js`

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js (hunk 2 above)
Comment (code-forger) [reply packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:8]: compiler is being passed in from webpack.

I have a sneaking suspicion what is happening here is we are making a class that looks like a function. But this is how the plugin I referenced to make this did it. So i just kept this done the same way as our other plugin

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js (hunk 1 above)
Comment (dogpatch626) [reply packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:15]: oh i completely missed the comment, makes sense

File: packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js (hunk 2 above)
Comment (dogpatch626) [reply packages/one-app-bundler/webpack/plugins/restrict-runtime-symbols.js:8]: funky stuff

**Review by dogpatch626 — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**